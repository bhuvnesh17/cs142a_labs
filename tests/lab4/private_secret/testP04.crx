// test nested loops

func main() : void {
  var a : int;
  var b : int;
  var c : int;

  while (((true))) {
    while (false) {
      while true {
        let a = 7;
      }
      let b = 8;
      if (true) {
        let a = 9;
      } else {
        if (false) {
          let b = 9;
        } else {
          while (false) {
            let b = 10;
          }
        }
        while (false) {
          let b = 3;
        }
      }
      while (true) {
        ::printInt(b);
      }
    }
    if (false) {
      ::printInt(c);
    }
  }
  if (true) {
    while (false) {
      ::printInt(a);
    }
  }
}
