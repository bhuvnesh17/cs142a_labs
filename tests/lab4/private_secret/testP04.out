ast.DeclarationList(3,1)
  ast.FunctionDefinition(3,1)[Symbol(main), []]
    ast.StatementList(4,3)
      ast.VariableDeclaration(4,3)[Symbol(a)]
      ast.VariableDeclaration(5,3)[Symbol(b)]
      ast.VariableDeclaration(6,3)[Symbol(c)]
      ast.WhileLoop(8,3)
        ast.LiteralBool(8,12)[TRUE]
        ast.StatementList(9,5)
          ast.WhileLoop(9,5)
            ast.LiteralBool(9,12)[FALSE]
            ast.StatementList(10,7)
              ast.WhileLoop(10,7)
                ast.LiteralBool(10,13)[TRUE]
                ast.StatementList(11,9)
                  ast.Assignment(11,9)
                    ast.AddressOf(11,13)[a]
                    ast.LiteralInt(11,17)[7]
              ast.Assignment(13,7)
                ast.AddressOf(13,11)[b]
                ast.LiteralInt(13,15)[8]
              ast.IfElseBranch(14,7)
                ast.LiteralBool(14,11)[TRUE]
                ast.StatementList(15,9)
                  ast.Assignment(15,9)
                    ast.AddressOf(15,13)[a]
                    ast.LiteralInt(15,17)[9]
                ast.StatementList(17,9)
                  ast.IfElseBranch(17,9)
                    ast.LiteralBool(17,13)[FALSE]
                    ast.StatementList(18,11)
                      ast.Assignment(18,11)
                        ast.AddressOf(18,15)[b]
                        ast.LiteralInt(18,19)[9]
                    ast.StatementList(20,11)
                      ast.WhileLoop(20,11)
                        ast.LiteralBool(20,18)[FALSE]
                        ast.StatementList(21,13)
                          ast.Assignment(21,13)
                            ast.AddressOf(21,17)[b]
                            ast.LiteralInt(21,21)[10]
                  ast.WhileLoop(24,9)
                    ast.LiteralBool(24,16)[FALSE]
                    ast.StatementList(25,11)
                      ast.Assignment(25,11)
                        ast.AddressOf(25,15)[b]
                        ast.LiteralInt(25,19)[3]
              ast.WhileLoop(28,7)
                ast.LiteralBool(28,14)[TRUE]
                ast.StatementList(29,9)
                  ast.Call(29,9)[Symbol(printInt)]
                    ast.ExpressionList(29,20)
                      ast.Dereference(29,20)
                        ast.AddressOf(29,20)[b]
          ast.IfElseBranch(32,5)
            ast.LiteralBool(32,9)[FALSE]
            ast.StatementList(33,7)
              ast.Call(33,7)[Symbol(printInt)]
                ast.ExpressionList(33,18)
                  ast.Dereference(33,18)
                    ast.AddressOf(33,18)[c]
            ast.StatementList(35,3)
      ast.IfElseBranch(36,3)
        ast.LiteralBool(36,7)[TRUE]
        ast.StatementList(37,5)
          ast.WhileLoop(37,5)
            ast.LiteralBool(37,12)[FALSE]
            ast.StatementList(38,7)
              ast.Call(38,7)[Symbol(printInt)]
                ast.ExpressionList(38,18)
                  ast.Dereference(38,18)
                    ast.AddressOf(38,18)[a]
        ast.StatementList(41,1)

